make/modules/java.desktop/lib/Awt2dLibraries.gmk

```diff
@@ -500,7 +500,7 @@ else
    # noexcept-type required for GCC 7 builds. Not required for GCC 8+.
    # expansion-to-defined required for GCC 9 builds. Not required for GCC 10+.
    HARFBUZZ_DISABLED_WARNINGS_CXX_gcc := class-memaccess noexcept-type expansion-to-defined dangling-reference
-   HARFBUZZ_DISABLED_WARNINGS_clang := missing-field-initializers range-loop-analysis
+   HARFBUZZ_DISABLED_WARNINGS_clang := missing-field-initializers
```

Comment:
Removed `range-loop-analysis` Disabled warning flag for clang. macOS build works with this change.

CI build works without array-bounds,  parentheses for HARFBUZZ_DISABLED_WARNINGS_gcc but since these flags were added recently to make build work on specific gcc versions retaining it - [PR#14996](https://github.com/openjdk/jdk/pull/14996/files) and [PR#13859](https://github.com/openjdk/jdk/pull/13859/files)

Reply:
Any specific reason why we need to remove `range-loop-analysis` flag as part of this update? Were there any build issues?

Reply:
@jayathirthrao

I removed `range-loop-analysis` as part of cleaning up the disabled warning section. I see that the following flag was added as part of https://github.com/openjdk/jdk/pull/993 to disable warnings for clang on Xcode 12.
Details here: https://github.com/openjdk/jdk/pull/993#issuecomment-722382063.  I suppose we need to add it back then?

Reply:
@prrace Makes sense! Thank you for the info about how this can be double-checked. I'll check and update the disabled section accordingly.

Reply:
With existing harfbuzz v7.2.0 (without the update) + Xcode 14 build passes which implies that the issue was resolved on Xcode 14 and not harfbuzz. Hence adding back `range-loop-analysis`.

Reply:
Since we know the reason for which this flag was added.
We should capture it in a comment, so that in future we will have faster way of analyzing it.

Reply:
Good point! Added the details as comment.